simpleflow/canvas.py

```diff
@@ -101,30 +102,47 @@ def append(self, *args, **kwargs):
             raise ValueError('{} should be an ActivityTask or an Activity'.format(args[0]))
 
     def submit(self, executor):
-        return GroupFuture(self.activities, executor)
+        return GroupFuture(self.activities, executor, self.max_parallel)
 
 
 class GroupFuture(futures.Future):
 
-    def __init__(self, activities, executor):
+    def __init__(self, activities, executor, max_parallel=None):
         super(GroupFuture, self).__init__()
         self.activities = activities
         self.futures = []
         self.executor = executor
+        self.max_parallel = max_parallel
+        self._count_pending_or_running = 0
+        self._lock = False
```

Comment:
I don't like the variable name, we don't understand what it does. Maybe: `self._futures_contain_all_activities`? Note that comparing the lengths may be enough, in which case we may be able to remove this variable and turn it into a method?